### ISSUE TITLE
Invalid output dir

### ISSUE THREAD

**Issue by elmobotix** [zdr1976/svbd#2]
I get:

NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\elmo/Videos/SafariBooks/Python for Beginners: Learn Python from Scratch'

**Comment by zdr1976** [comment]
Hi,

I'm no more using this scrip as the [youtube-dl](https://ytdl-org.github.io/youtube-dl/index.html) can do it by its own. It can also download videos from another popular site like udemy an so. Please have a look.

**Comment by elmobotix** [comment]
Hi, Ah OK. Can YouTube-dl manage full batch downloads so I can get a full course rather than single videos from Oreilly?

**Comment by zdr1976** [comment]
Yes this tool can download whole video course.

**Comment by elmobotix** [comment]
Great many thanks. I’ve been trying for ages to get full courses but I’m
not good at coding etc so I find it a bit challenging. Do you know if
anyone has written a guide on how to do it.

**Comment by zdr1976** [comment]
So here you are example how to download via `youtube-dl`

```bash
youtube-dl -u your@account.com -o '%(playlist_title)s/%(playlist_index)s-%(title)s.%(ext)s' -f 'bestvideo[height<=720]+bestaudio/best[height<=720]/worst' https://learning.oreilly.com/videos/node-js-api-masterclass/9781800569638/
```

You can adjust quality and output as you need. Check manual pages for more options. This is one i use.

**Comment by elmobotix** [comment]
So, I assume I can either use the same folder that I use for getting books
from safari, or I simply make a new folder. Ive just tried both of those
ways and have but my email in the script (no password!! - or do I put
":mypassword" after the email address?)

I get an error anyway.
[image: image.png]

**Comment by elmobotix** [comment]
[image: image.png]I did use -u before the email address.

**Comment by elmobotix** [comment]
Doh, just looked on Github, theres a repository to get first right!

**Comment by zdr1976** [comment]
`-u` is user name on oreilly.com it's email address as your account.

**Comment by elmobotix** [comment]
Yeah sorry I understood that part. Do I need the youtube-dl repository from
Github https://github.com/ytdl-org/youtube-dl.git ?

**Comment by zdr1976** [comment]
Just follow the installation instructions [here](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#installation).

**Comment by elmobotix** [comment]
Hi,

Yes once I realised there was a specific youtude-dl repository I got it
working. Thankyou for helping. Im grabbing some videos right now but I just
need to work out how to stop it from adding characters to the file name
such as...

   1. Lesson 1 - Getting Started with IPv4 Addresses*-0_cs1sxl8j*.mp4
   (where  "-0_cs1sxl8j"  has been added)

**Comment by zdr1976** [comment]
In the example above i'm using `-o '%(playlist_title)s/%(playlist_index)s-%(title)s.%(ext)s'` to name the directory and files. So just modify it to your liking.

**Comment by elmobotix** [comment]
Perfect!! - thankyou sooo much fore your help ...lifesaver.

**Comment by zdr1976** [comment]
You're welcome.

**Comment by elmobotix** [comment]
Hi again

Do you know how difficult it would be to tweak the .py file so that when using  --login option you get prompted for email address, and then the book ID

Steve

**Comment by elmobotix** [comment]
Hi again

Do you the books downloader too? 

I use it but would like to know how I could tweak the .py to prompt for book ID as well as email addy and password.

Any ideas
Steve

**Comment by zdr1976** [comment]
You can tweak it as much as you want. For argument parsing you can use [argparse](https://docs.python.org/3.8/library/argparse.html) python module.

**Comment by elmobotix** [comment]
I’m literally on day 1 of learning python but I will certainly hit it a go.

Thanks
